bot/cogs/hiphopify.py

```diff
@@ -0,0 +1,184 @@
+import logging
+import random
+
+from discord import Colour, Embed, Member
+from discord.ext.commands import AutoShardedBot, Context, command
+
+from bot.constants import (
+    ADMIN_ROLE, MODERATOR_ROLE, MOD_LOG_CHANNEL,
+    NEGATIVE_REPLIES, OWNER_ROLE, POSITIVE_REPLIES,
+    SITE_API_HIPHOPIFY_URL, SITE_API_KEY
+)
+from bot.decorators import with_role
+
+log = logging.getLogger(__name__)
+
+
+class Hiphopify:
+    """
+    A set of commands to moderate terrible nicknames.
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+        self.headers = {"X-API-KEY": SITE_API_KEY}
+
+    async def on_member_update(self, before, after):
+        """
+        This event will trigger when someone changes their name.
+        At this point we will look up the user in our database and check
+        whether they are allowed o change their names, or if they are in
+        hiphop-prison. If they are not allowed, we will change it back.
+        :return:
+        """
+
+        if before.display_name == after.display_name:
+            return  # User didn't change their nickname. Abort!
+
+        log.debug(
+            f"{before.display_name} is trying to change their nickname to {after.display_name}. "
+            f"Checking if the user is in hiphop-prison..."
+        )
+
+        response = await self.bot.http_session.get(
+            SITE_API_HIPHOPIFY_URL,
+            headers=self.headers,
+            params={"user_id": str(before.id)}
+        )
+
+        response = await response.json()
+
+        if response:
+
+            if after.display_name == response.get("forced_nick"):
+                return  # Nick change was triggered by this event. Ignore.
+
+            log.warning(
```

Comment:
Under our previous discussion regarding error levels we decided that the definition for the warning level was
> Someone is interacting with the application in an unexpected way or the application is responding in an unexpected way, but without causing an error.

I think that either DEBUG or INFO would be more suitable for this.

Reply:
agreed.